### PR TITLE
Prevent asterism updates for executed observations, add undo/redo

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: explore/src/clue/scala/queries/common/AsterismQueriesGQL.scala [hunk 1]
```diff
@@ -20,4 +20,25 @@ object AsterismQueriesGQL {
       }
     """
   }
+
+  // Group these 2 mutations together so that the targetEdit and observationEdit subscriptions come
```
Comment (cquiroz) [inline explore/src/clue/scala/queries/common/AsterismQueriesGQL.scala:24]: Is this guaranteed to work?

File: explore/src/main/scala/explore/tabs/AsterismEditorTile.scala [hunk 2]
```diff
@@ -59,6 +59,7 @@ object AsterismEditorTile:
     backButton:        Option[VdomNode] = none
   )(using FetchClient[IO, ObservationDB], Logger[IO]): Tile = {
     // Save the time here. this works for the obs and target tabs
+    // It's OK to save the viz time for executed observations, I think.
     val vizTimeView =
       vizTime.withOnMod(t => ObsQueries.updateVisualizationTime[IO](obsIds.toList, t).runAsync)
```
Comment (cquiroz) [inline explore/src/main/scala/explore/tabs/AsterismEditorTile.scala:64]: This needs an update I think

**Review by cquiroz — APPROVED** [review]
This is quite an undertaking! thanks
LGTM

File: explore/src/clue/scala/queries/common/AsterismQueriesGQL.scala (hunk 1 above)
Comment (toddburnside) [reply explore/src/clue/scala/queries/common/AsterismQueriesGQL.scala:24]: Well, we reduce the transit time involved, but they still aren't real close together. That timing depends on when the database updates happen for the 2 mutations, I guess. Still seemed better than doing them individually. 🤷‍♂️

File: explore/src/main/scala/explore/tabs/AsterismEditorTile.scala (hunk 2 above)
Comment (toddburnside) [reply explore/src/main/scala/explore/tabs/AsterismEditorTile.scala:64]: Actually, it looks like it is still called `ObsQueries.updateVisualizationTime` in your PR. 😄 So, I don't think this needs to change. Unless there is another reason.

**Pull request merged**